rtfbook/chapter_rdm/metadata.md

```diff
@@ -13,44 +13,178 @@ kernelspec:
+
+```{note}
+To create `unitpackages`, the structure for the fields must be followed. The `name` for each field must match a field/column name in the CSV (datafile).
 ```
 
 ```{code-cell} ipython3
 :tags: [remove-input]
 from IPython.display import Code
 
-Code(filename='../files/data/data.csv.meta.yaml', language='yaml')
+Code(filename='../files/yaml_templates/figure_description.yaml', language='yaml')
+```
+
+**experimental**: Details on the equipment and experimental procedures applied to the system that is described below.
+
+```{code-cell} ipython3
+:tags: [remove-input]
+from IPython.display import Code
+
+Code(filename='../files/yaml_templates/experimental.yaml', language='yaml')
+```
+
+**system**: Details on the system on which the experiment is performed, including anything that is in contact with or any eternal parameters that can have an impact on the measurement. Consider putting a banana in a beaker to study how its color changes when the external parameters, such as the atmosphere, temperature and light intensity/source.
+Also, include information that does not seem to be relevant at the time of the data acquisition.
+
+```{note}
+This part is presumably the most important one. With increasing amount of information, more correlations can be drawn and possible issues can be found.
+It also allows elucidating possible variations in future measurements. For example, assume a new student uses a new beaker in a new study on the same system, but does not receive the same results. If the beaker would be the problem, but its properties (glass, thickness, supplier, an internal ID, ...) were not recorded properly, one could not so easily pin down the origin of the problem.
+```
+
+```{code-cell} ipython3
+:tags: [remove-input]
+from IPython.display import Code
+
+Code(filename='../files/yaml_templates/system.yaml', language='yaml')
+```
+
+An extensive example of a more complete YAML template for an electrochemical system can be found [here](https://github.com/echemdb/metadata-schema/blob/main/examples/file_schemas/autotag.yaml).
+
+A complete YSML would contain all of the above (and other custom) categories.
+
+```{code-cell} ipython3
+:tags: [remove-input]
+from IPython.display import Code
+
+Code(filename='../files/yaml_templates/measurement.yaml', language='yaml')
 ```
+
+
+## Loading templates
+
+The YAML files can be loaded as a Python dictionary.
+
+```{code-cell} ipython3
+import yaml
+
+metadata = yaml.load('../data/files/data.csv.meta.yaml', Loader=yaml.SafeLoader)
+```
+
+## Annotating data automatically
+
+When (measurement) files are created in the file system, preferably the content of the YAML template is directly associated with the measurement file.
+This can be achieved by monitoring the file system for newly created files. Following we illustrate the approach by using the Python watchdog module.
```

Comment:
```suggestion
This can be achieved by monitoring the file system for newly created files. Following, we illustrate the approach by using the Python watchdog package.
```